tests/samplers_tests/test_samplers.py

```diff
@@ -549,3 +550,27 @@ def after_trial(
     study.tell(study.ask(), 1.0)
 
     assert n_calls == 1
+
+
+@parametrize_sampler
+def test_sample_single_distribution(sampler_class: Callable[[], BaseSampler]) -> None:
+
+    relative_search_space = {
+        "a": UniformDistribution(low=1.0, high=1.0),
+        "b": LogUniformDistribution(low=1.0, high=1.0),
+        "c": DiscreteUniformDistribution(low=1.0, high=1.0, q=1.0),
+        "d": IntUniformDistribution(low=1, high=1),
+        "e": IntLogUniformDistribution(low=1, high=1),
+        "f": CategoricalDistribution([1]),
+    }
+
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore", optuna.exceptions.ExperimentalWarning)
+        sampler = sampler_class()
+    study = optuna.study.create_study(sampler=sampler)
+
+    for _ in range(100):
+        trial = study.ask(fixed_distributions=relative_search_space)
```

Comment:
[question]

In my environment, the test failed with `SkoptSampler` (with version 0.8.1).
Is this expected?


<details>

```
tests/samplers_tests/test_samplers.py ....F..                                                                                                                            [100%]

=================================================================================== FAILURES ===================================================================================
__________________________________________________________________ test_sample_single_distribution[<lambda>3] __________________________________________________________________

sampler_class = <function <lambda> at 0x137fbbee0>

    @parametrize_sampler
    def test_sample_single_distribution(sampler_class: Callable[[], BaseSampler]) -> None:

        relative_search_space = {
            "a": UniformDistribution(low=1.0, high=1.0),
            "b": LogUniformDistribution(low=1.0, high=1.0),
            "c": DiscreteUniformDistribution(low=1.0, high=1.0, q=1.0),
            "d": IntUniformDistribution(low=1, high=1),
            "e": IntLogUniformDistribution(low=1, high=1),
            "f": CategoricalDistribution([1]),
        }

        with warnings.catch_warnings():
            warnings.simplefilter("ignore", optuna.exceptions.ExperimentalWarning)
            sampler = sampler_class()
        study = optuna.study.create_study(sampler=sampler)

        for _ in range(100):
>           trial = study.ask(fixed_distributions=relative_search_space)

tests/samplers_tests/test_samplers.py:573:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
optuna/study.py:487: in ask
    trial = trial_module.Trial(self, trial_id)
optuna/trial/_trial.py:57: in __init__
    self._init_relative_params()
optuna/trial/_trial.py:66: in _init_relative_params
    self.relative_params = self.study.sampler.sample_relative(
optuna/integration/skopt.py:171: in sample_relative
    optimizer.tell(study, complete_trials)
optuna/integration/skopt.py:288: in tell
    self._optimizer.tell(xs, ys)
venv/lib/python3.9/site-packages/skopt/optimizer/optimizer.py:493: in tell
    return self._tell(x, y, fit=fit)
venv/lib/python3.9/site-packages/skopt/optimizer/optimizer.py:536: in _tell
    est.fit(self.space.transform(self.Xi), self.yi)
venv/lib/python3.9/site-packages/skopt/learning/gaussian_process/gpr.py:195: in fit
    super(GaussianProcessRegressor, self).fit(X, y)
venv/lib/python3.9/site-packages/sklearn/gaussian_process/_gpr.py:235: in fit
    optima = [(self._constrained_optimization(obj_func,
venv/lib/python3.9/site-packages/sklearn/gaussian_process/_gpr.py:506: in _constrained_optimization
    opt_res = scipy.optimize.minimize(
venv/lib/python3.9/site-packages/scipy/optimize/_minimize.py:619: in minimize
    return _minimize_lbfgsb(fun, x0, args, jac, bounds,
venv/lib/python3.9/site-packages/scipy/optimize/lbfgsb.py:306: in _minimize_lbfgsb
    sf = _prepare_scalar_function(fun, x0, jac=jac, args=args, epsilon=eps,
venv/lib/python3.9/site-packages/scipy/optimize/optimize.py:261: in _prepare_scalar_function
    sf = ScalarFunction(fun, x0, args, grad, hess,
venv/lib/python3.9/site-packages/scipy/optimize/_differentiable_functions.py:136: in __init__
    self._update_fun()
venv/lib/python3.9/site-packages/scipy/optimize/_differentiable_functions.py:226: in _update_fun
    self._update_fun_impl()
venv/lib/python3.9/site-packages/scipy/optimize/_differentiable_functions.py:133: in update_fun
    self.f = fun_wrapped(self.x)
venv/lib/python3.9/site-packages/scipy/optimize/_differentiable_functions.py:130: in fun_wrapped
    return fun(x, *args)
venv/lib/python3.9/site-packages/scipy/optimize/optimize.py:74: in __call__
    self._compute_if_needed(x, *args)
venv/lib/python3.9/site-packages/scipy/optimize/optimize.py:68: in _compute_if_needed
    fg = self.fun(x, *args)
venv/lib/python3.9/site-packages/sklearn/gaussian_process/_gpr.py:227: in obj_func
    lml, grad = self.log_marginal_likelihood(
venv/lib/python3.9/site-packages/sklearn/gaussian_process/_gpr.py:481: in log_marginal_likelihood
    alpha = cho_solve((L, True), y_train)  # Line 3
venv/lib/python3.9/site-packages/scipy/linalg/decomp_cholesky.py:194: in cho_solve
    b1 = asarray_chkfinite(b)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

a = array([[nan]]), dtype = None, order = None

    @set_module('numpy')
    def asarray_chkfinite(a, dtype=None, order=None):
        """Convert the input to an array, checking for NaNs or Infs.

        Parameters
        ----------
        a : array_like
            Input data, in any form that can be converted to an array.  This
            includes lists, lists of tuples, tuples, tuples of tuples, tuples
            of lists and ndarrays.  Success requires no NaNs or Infs.
        dtype : data-type, optional
            By default, the data-type is inferred from the input data.
        order : {'C', 'F', 'A', 'K'}, optional
            Memory layout.  'A' and 'K' depend on the order of input array a.
            'C' row-major (C-style),
            'F' column-major (Fortran-style) memory representation.
            'A' (any) means 'F' if `a` is Fortran contiguous, 'C' otherwise
            'K' (keep) preserve input order
            Defaults to 'C'.

        Returns
        -------
        out : ndarray
            Array interpretation of `a`.  No copy is performed if the input
            is already an ndarray.  If `a` is a subclass of ndarray, a base
            class ndarray is returned.

        Raises
        ------
        ValueError
            Raises ValueError if `a` contains NaN (Not a Number) or Inf (Infinity).

        See Also
        --------
        asarray : Create and array.
        asanyarray : Similar function which passes through subclasses.
        ascontiguousarray : Convert input to a contiguous array.
        asfarray : Convert input to a floating point ndarray.
        asfortranarray : Convert input to an ndarray with column-major
                         memory order.
        fromiter : Create an array from an iterator.
        fromfunction : Construct an array by executing a function on grid
                       positions.

        Examples
        --------
        Convert a list into an array.  If all elements are finite
        ``asarray_chkfinite`` is identical to ``asarray``.

        >>> a = [1, 2]
        >>> np.asarray_chkfinite(a, dtype=float)
        array([1., 2.])

        Raises ValueError if array_like contains Nans or Infs.

        >>> a = [1, 2, np.inf]
        >>> try:
        ...     np.asarray_chkfinite(a)
        ... except ValueError:
        ...     print('ValueError')
        ...
        ValueError

        """
        a = asarray(a, dtype=dtype, order=order)
        if a.dtype.char in typecodes['AllFloat'] and not np.isfinite(a).all():
>           raise ValueError(
                "array must not contain infs or NaNs")
E           ValueError: array must not contain infs or NaNs

venv/lib/python3.9/site-packages/numpy/lib/function_base.py:488: ValueError
```

</details>

Reply:
@himkt  Could you check that your scikit-learn version is the latest stable `0.24.2`. Otherwise, skopt sampler does not work as you mentioned.
ref: https://github.com/optuna/optuna/pull/1890#issuecomment-702686331

Reply:
Thanks for the report @himkt and the reply @nzw0301. As @nzw0301 commented, the `SkoptSampler` does not support the latest version. I have the same error with the latest version.

Reply:
Thank you, I confirmed tests passed with scikit-learn 0.24.2.